src/Enterspeed.Source.UmbracoCms/Controllers/Api/DashboardApiController.cs

```diff
@@ -238,6 +240,41 @@ public IActionResult ClearPendingJobs()
             }
         }
 
+        [HttpPost]
+        public ActionResult DeleteFailedJobs()
+        {
+            using (_scopeProvider.CreateScope(autoComplete: true))
+            {
+                var failedJobs = _enterspeedJobRepository.GetFailedJobs();
+                if (failedJobs != null && failedJobs.Any())
+                {
+                    _enterspeedJobRepository.Delete(failedJobs.Select(fj => fj.Id).ToList());
+                }
+            }
+
+            return Ok(new ApiResponse()
+            {
+                IsSuccess = true
+            });
+        }
+
+        [HttpPost]
+        public ActionResult DeleteJobs(JobIdsToDelete jobIdsToDelete)
+        {
+            using (_scopeProvider.CreateScope(autoComplete: true))
+            {
+                if (jobIdsToDelete != null && jobIdsToDelete.Ids.Any())
```

Comment:
Should we get the jobs first, to check if they actually still exists? - I am unsure what will happen if someone select a job, that in the meantime gets processed succesfully